x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -848,7 +848,12 @@ export const LensTopNavMenu = ({
 
   const onSavedQueryUpdatedWrapped = useCallback(
     (newSavedQuery) => {
-      const savedQueryFilters = newSavedQuery.attributes.filters || [];
+      // If the user tries to load the same saved query that is already loaded,
+      // we will receive the same object reference which was previously frozen
+      // by Redux Toolkit. `filterManager.setFilters` will then try to modify
+      // the query's filters, which will throw an error. To avoid this, we need
+      // to clone the filters before passing them to `filterManager.setFilters`.
+      const savedQueryFilters = cloneDeep(newSavedQuery.attributes.filters || []);
```

Comment:
ℹ️ This bug was already present in `main`, but was never caught in any functional tests until I added this check to `test/functional/services/saved_query_management_component.ts`, which caused it to start triggering test failures: https://github.com/elastic/kibana/blob/119ca6d36d044f3ec0997a409fc119532ecc448e/test/functional/services/saved_query_management_component.ts#L90-L92

Reply:
great catch!